presentation-compiler/test/dotty/tools/pc/tests/edit/InsertInferredMethodSuite.scala

```diff
@@ -0,0 +1,502 @@
+       |
+       |    def otherMethod(arg0: User, arg1: Int) = ???
+       |    otherMethod(User(1), 1)
+       |}
+       |""".stripMargin
+    )
+
+  @Test def `custom-type-advanced-2` =
+    checkEdit(
+    """|
+       |trait Main {
+       |    def method1(b: Double, s : String) = 123
+       |
+       |    case class User(i : Int)
+       |
+       |    <<otherMethod>>(List(Set(User(1))), Map("1" -> 1))
+       |}
+       |
+       |""".stripMargin,
+    """|trait Main {
+       |    def method1(b: Double, s : String) = 123
+       |
+       |    case class User(i : Int)
+       |
+       |    def otherMethod(arg0: List[Set[User]], arg1: Map[String, Int]) = ???
+       |    otherMethod(List(Set(User(1))), Map("1" -> 1))
+       |}
+       |""".stripMargin
+    )
+
+  @Test def `with-imports` =
+    checkEdit(
+    """|import java.nio.file.Files
+       |
+       |trait Main {
+       |  def main() = {
+       |    def method1(s : String) = 123
+       |      val path = Files.createTempDirectory("")
+       |      method1(<<otherMethod>>(path))
+       |    }
+       |}
+       |
+       |""".stripMargin,
+    """|import java.nio.file.Files
+       |import java.nio.file.Path
+       |
+       |trait Main {
+       |  def main() = {
+       |    def method1(s : String) = 123
+       |      val path = Files.createTempDirectory("")
+       |      def otherMethod(arg0: Path): String = ???
+       |      method1(otherMethod(path))
+       |    }
+       |}
+       |""".stripMargin
+    )
+
+  @Test def `val-definition` =
+    checkEdit(
+      """|
+        |trait Main {
+        |  val result: String = <<nonExistent>>(42, "hello")
+        |}
+        |
+        |""".stripMargin,
+      """|trait Main {
+        |  def nonExistent(arg0: Int, arg1: String): String = ???
+        |  val result: String = nonExistent(42, "hello")
+        |}
+        |""".stripMargin
+    )
+
+  @Test def `val-definition-no-args` =
+    checkEdit(
+      """|
+        |trait Main {
+        |  val result: Int = <<getValue>>
+        |}
+        |
+        |""".stripMargin,
+      """|trait Main {
+        |  def getValue: Int = ???
+        |  val result: Int = getValue
+        |}
+        |""".stripMargin
+    )
+
+  @Test def `lambda-expression` =
+    checkEdit(
+      """|
+        |trait Main {
+        |  val list = List(1, 2, 3)
+        |  list.map(<<transform>>)
+        |}
+        |
+        |""".stripMargin,
+      """|trait Main {
+        |  val list = List(1, 2, 3)
+        |  def transform(arg0: Int) = ???
+        |  list.map(transform)
+        |}
+        |""".stripMargin
+    )
+
+  @Test def `lambda-expression-2` =
+    checkEdit(
+      """|
+        |trait Main {
+        |  val list = List(1, 2, 3)
+        |  list.map(<<transform>>(10, "test"))
+        |}
+        |
+        |""".stripMargin,
+      """|trait Main {
+        |  val list = List(1, 2, 3)
+        |  def transform(arg0: Int, arg1: String, arg2: Int) = ???
```

Comment:
```suggestion
        |  def transform(arg0: Int, arg1: String)(arg2: Int) = ???
```

otherwise, it doesn't compile